accepted/2023/simplify-output-paths.md

```diff
@@ -0,0 +1,140 @@
+- Provide simple and consistent output paths
+- Avoid cases where one output path is nested inside of another one
+- Avoid excessively deep directory heirarchies
+- Allow output path customization when needed
+
+## Motivation
+
+Some developers would like to put all the output for their solution or repo [under a single folder](https://github.com/dotnet/sdk/issues/867).  This would make it easier to delete all of the output, as MSBuild-based clean only deletes output for a single configuration, and doesn't delete everything from the intermediate output folder, as that's where it keeps track of what was produced by the build and hence what should be deleted in a clean.  It would support building with [read only](https://github.com/dotnet/sdk/issues/887) [source trees](https://github.com/dotnet/designs/pull/281#discussion_r1072949097).  It could also make it easier to set up CI and deployment.
+
+The current output paths for .NET projects are rather complicated and inconsistent.  The default output path for a .NET project includes folders for the `TargetFramework` and the `RuntimeIdentifier` (if it is set).  Additionally, the publish output goes in a `publish` subfolder of the output folder.  The resulting paths are as follows:
+
+- `bin\<Configuration>\<TargetFramework>` - Build output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>` - Build output with `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\publish` - Publish output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>\publish` - Publish output with `RuntimeIdentifier`
+
+## Proposed behavior
+
+We will introduce a new output path format, called the "artifacts" output format after the default folder name it will use.  By convention, the artifacts folder will be located at the solution or repo level rather within each project folder.
+
+The recommended way to opt in to the new output path format will be to set the `UseArtifactsOutput` property to `true` in a `Directory.Build.props` file.
+
+The artifacts folder will by default be named `.artifacts`.  The convention we will use to determine which folder the artifacts folder should go in will be:
+
+- If a `Directory.Build.props` file was used in the build, put the artifacts folder in the same folder where `Directory.Build.props` was found, which will typically be at the root of a repo.  Since MSBuild already probes for `Directory.Build.props`, we should be able to use this folder for the artifacts with minimal perf impact
+- If a `Directory.Build.props` file was not found, then put the artifacts folder in the project folder.
+
+To override the default artifacts folder path, the `ArtifactsPath` property can be used.  If `ArtifactsPath` is set, then `UseArtifactsOutput` will default to `true` and it won't be necessary to set both properties.  `ArtifactsPath` should normally be set in a `Directory.Build.props` file.  For example:
```

Comment:
Do we need these two separate properties? Or could we only have `ArtifactsPath`?

Maybe that would also make the discussion around the default folder name easier, if we use setting the 
`ArtifactsPath` as the opt-in gesture then the user can provide the preferred name (we could still recommend some default name in docs of course).